pkg/apis/duck/v1beta1/subscribable_types.go

```diff
@@ -0,0 +1,177 @@
+}
+
+// Subscribable is an Implementable "duck type".
+var _ duck.Implementable = (*Subscribable)(nil)
+
+// SubscriberSpec defines a single subscriber to a Subscribable.
+// Ref is a reference to the Subscription this SubscriberSpec was created for
+// SubscriberURI is the endpoint for the subscriber
+// ReplyURI is the endpoint for the reply
+// At least one of SubscriberURI and ReplyURI must be present
+type SubscriberSpec struct {
+	// UID is used to understand the origin of the subscriber.
+	// +optional
+	UID types.UID `json:"uid,omitempty"`
+	// Generation of the origin of the subscriber with uid:UID.
+	// +optional
+	Generation int64 `json:"generation,omitempty"`
+	// +optional
+	SubscriberURI *apis.URL `json:"subscriberUri,omitempty"`
+	// +optional
+	ReplyURI *apis.URL `json:"replyUri,omitempty"`
+	// +optional
+	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkUri,omitempty"`
+}
+
+// SubscribableStatus is the schema for the subscribable's status portion of the status
+// section of the resource.
+type SubscribableStatus struct {
+	// This is the list of subscription's statuses for this channel.
+	// +patchMergeKey=uid
+	// +patchStrategy=merge
+	Subscribers []SubscriberStatus `json:"subscribers,omitempty" patchStrategy:"merge" patchMergeKey:"uid"`
+}
+
+// SubscriberStatus defines the status of a single subscriber to a Channel.
+type SubscriberStatus struct {
+	// UID is used to understand the origin of the subscriber.
+	// +optional
+	UID types.UID `json:"uid,omitempty"`
+	// Generation of the origin of the subscriber with uid:UID.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// Status of the subscriber.
+	Ready corev1.ConditionStatus `json:"ready,omitempty"`
+	// A human readable message indicating details of Ready status.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SubscribableType is a skeleton type wrapping Subscribable in the manner we expect resource writers
+// defining compatible resources to embed it. We will typically use this type to deserialize
+// SubscribableType ObjectReferences and access the Subscription data.  This is not a real resource.
+type SubscribableType struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// SubscribableTypeSpec is the part where Subscribable object is
+	// configured as to be compatible with Subscribable contract.
+	Spec SubscribableTypeSpec `json:"spec"`
+
+	// SubscribableTypeStatus is the part where SubscribableStatus object is
+	// configured as to be compatible with Subscribable contract.
+	Status SubscribableTypeStatus `json:"status"`
+}
+
+// SubscribableTypeSpec shows how we expect folks to embed Subscribable in their Spec field.
+type SubscribableTypeSpec struct {
+	Subscribable *Subscribable `json:"subscribable,omitempty"`
```

Comment:
I always felt that `subscribable` is redundant. Why not just:
```yaml
spec:
  subscribers: 
  - ...
```

Reply:
ack :)